macros/g4simulations/Fun4All_G4_EICDetector.C

```diff
@@ -192,7 +196,7 @@ int Fun4All_G4_EICDetector(
 
   // establish the geometry and reconstruction setup
   gROOT->LoadMacro("G4Setup_EICDetector.C");
-  G4Init(do_tracking,do_cemc,do_hcalin,do_magnet,do_hcalout,do_pipe,do_plugdoor,do_FEMC,do_FHCAL,do_EEMC,do_DIRC,do_RICH,do_Aerogel);
+  G4Init();
```

Comment:
Hi Jin,

it's still very rough to show were I think things should go (you caught me having checked in what I was working with - that PR is testing the waters and should definitely not be merged in the current form). Everything is open for discussion.
I took the EIC macro since there is a lot of interest in flexibility (and I won't break sPHENIX) but this should also be applied to sPHENIX then. The underlying subsystem macros are mostly ready.

I did not want to break existing privately changed macros (that's why the overlap is still there). Those macros will not get the close fit black hole or increase in the world volume but should stay functional - at least for a while - to make this less disruptive. 

The main problem will be the tracking macro, we have been very lax with global variables. Since all macros are #included the scope of every variable declared outside of the functions extends to the whole project. That's why they should move into distinct namespaces.

We can move this to the DST reader but I am not sure what to do about local flags (like do_cemc_cell) which stay local in the Fun4All macro. Replacing those with an enormous list of booleans in the Enable namespace sounds counter productive. Given that new developers are coming I want to avoid single large files which everyone has to modify. Probably separate namespaces loaded with the G4 macro is the way to go? That would also take care of subsystem specific flags like the eic beam pipe (this would blow up if you load the wrong macro but I would see this as a failsafe).

The overlap check for the blackhole is essential - you can easily place this inside your detector and you'll never figure out why you are loosing hits (yes that happened to me which made me think about this). I'd rather look into suppressing the messages - even if it takes a G4 hack (same for the gdml, these messages are hardcoded - not sure who came up with that).